websites.py

```diff
@@ -0,0 +1,10 @@
+patterns = {
+	'pinterest': r'^[a-zA-Z0-9_]{3,15}$',
```

Comment:
I think it is more useful to blow this regex into parts in the data here.  have min and max lengths as separate fields, and store a "valid characters" list containing `a-zA-Z0-9_`.

Then at start build the compiled regexes using `^[{valid_characters}]{min,max}$'`.

`min` of 1 is implicit.

Error if max isnt provided.